consent-ws/src/main/java/org/broadinstitute/consent/http/service/NihServiceAPI.java

```diff
@@ -0,0 +1,83 @@
+package org.broadinstitute.consent.http.service;
+
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.UnsupportedJwtException;
+import io.jsonwebtoken.io.DecodingException;
+import io.jsonwebtoken.security.Keys;
+import io.jsonwebtoken.security.SignatureException;
+import java.security.Key;
+
+import org.broadinstitute.consent.http.configurations.NihConfiguration;
+import org.broadinstitute.consent.http.enumeration.ResearcherFields;
+import org.broadinstitute.consent.http.models.ResearcherProperty;
+import org.broadinstitute.consent.http.service.users.handler.ResearcherAPI;
+
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Calendar;
+import java.util.Date;
+
+public class NihServiceAPI implements NihAuthApi {
+
+    private NihConfiguration nihConfiguration;
+    private ResearcherAPI researcherAPI;
+
+    public NihServiceAPI(NihConfiguration nihConfiguration, ResearcherAPI researcherApi) {
+        this.researcherAPI = researcherApi;
+        this.nihConfiguration = nihConfiguration;
+    }
+
+    @Override
+    public Map<String, String> authenticateNih(String jwt, Integer userId) throws SignatureException, MalformedJwtException, UnsupportedJwtException, DecodingException {
+
+        Key key = Keys.hmacShaKeyFor(nihConfiguration.getSigningSecret());
+
+        String nihUserName = Jwts.parser()
+                .setSigningKey(key)
+                .parsePlaintextJws(jwt).getBody();
```

Comment:
See my comment about exceptions above. I think this block should catch and log (probably at the warning level for client-provided `jwt` tokens) all of the exceptions possible with some explanatory language for each.

Reply:
See the [quickstart example](https://github.com/jwtk/jjwt#quickstart): 
```
try {
    Jwts.parser().setSigningKey(key).parseClaimsJws(compactJws);
    //OK, we can trust this JWT
} catch (JwtException e) {
    //don't trust the JWT!
}
```
In our implementation, `parsePlaintextJws` is where we care about catching any kind of error. This is where we'll find any kind of security problem with the token. These should be logged as severe (with some more info about user, request, etc.,) so we can follow up with those users via our normal communications process.